controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -163,40 +163,48 @@ void client_steering_task::steer_sta()
         /*
         * send disallow to all others
         */
-        agent_mac = database.get_node_parent_ire(hostap);
-        if (!cmdu_tx.create(0,
-                            ieee1905_1::eMessageType::CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE)) {
-            LOG(ERROR)
-                << "cmdu creation of type CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE, has failed";
-            return;
-        }
-
-        auto association_control_block_request_tlv =
-            cmdu_tx.addClass<wfa_map::tlvClientAssociationControlRequest>();
-        if (!association_control_block_request_tlv) {
-            LOG(ERROR) << "addClass wfa_map::tlvClientAssociationControlRequest failed";
-            return;
+        auto hostap_vap_lits = database.get_hostap_vap_list(hostap);
+        auto ssid            = database.get_hostap_ssid(target_bssid);
+        for (auto &hostap_vap : hostap_vap_lits) {
+            if (hostap_vap.second.ssid == ssid) {
+
+                agent_mac = database.get_node_parent_ire(hostap);
+                if (!cmdu_tx.create(
+                        0, ieee1905_1::eMessageType::CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE)) {
+                    LOG(ERROR) << "cmdu creation of type "
+                                  "CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE, has failed";
+                    return;
+                }
+
+                auto association_control_block_request_tlv =
+                    cmdu_tx.addClass<wfa_map::tlvClientAssociationControlRequest>();
+                if (!association_control_block_request_tlv) {
+                    LOG(ERROR) << "addClass wfa_map::tlvClientAssociationControlRequest failed";
+                    return;
+                }
+
+                association_control_block_request_tlv->association_control() =
+                    wfa_map::tlvClientAssociationControlRequest::BLOCK;
+                association_control_block_request_tlv->validity_period_sec() =
+                    steering_wait_time_ms / 1000;
+                association_control_block_request_tlv->alloc_sta_list();
+                auto sta_list_block         = association_control_block_request_tlv->sta_list(0);
+                std::get<1>(sta_list_block) = network_utils::mac_from_string(sta_mac);
+                association_control_block_request_tlv->bssid_to_block_client() =
+                    network_utils::mac_from_string(hostap_vap.second.mac);
```

Comment:
Please keep the order of setting TLV fields to the same order as defined in the structure, because in some cases the tlvf is sensitive to the order. It was correct in the original.